src/app/mesh/qgsmeshdatasetgrouptreeview.cpp

```diff
@@ -588,13 +593,36 @@ void QgsMeshDatasetGroupTreeView::contextMenuEvent( QContextMenuEvent *event )
 
 void QgsMeshDatasetGroupTreeView::removeCurrentItem()
 {
-  if ( QMessageBox::question( this, tr( "Remove Dataset Group" ), tr( "Remove dataset group?" ) ) == QMessageBox::Ok )
+  QgsMeshDatasetGroupTreeItem *item = mModel->datasetGroupTreeItem( currentIndex() );
+
+  if ( item )
+  {
+    QList<int> dependencies = item->groupIndexDependencies();
+    if ( !dependencies.isEmpty() )
+    {
+      QString varList;
+      for ( int dependentGroupIndex : dependencies )
+      {
+        QgsMeshDatasetGroupTreeItem *item = mModel->datasetGroupTreeItem( dependentGroupIndex );
+        if ( item )
+        {
+          varList.append( item->name() );
+          varList.append( QStringLiteral( "\n" ) );
+        }
+      }
+      QMessageBox::information( this, tr( "Remove Dataset Group" ), tr( "This dataset group can be removed because it has the following dependencies :\n%1" )
```

Comment:
can -> can't  ?

Also, is it correct that `groupIndexDependencies()` returns other groups that depend on this group? (or is it the other way around: that this group depends on some other groups?). In case of the former, I would maybe reformulate "because it has the following dependencies:" to something like "because other dataset groups depend on it:"

Reply:
first case.